packages/in_app_purchase/ios/Classes/InAppPurchasePlugin.m

```diff
@@ -35,4 +44,49 @@ - (void)canMakePayments:(FlutterResult)result {
   result([NSNumber numberWithBool:[SKPaymentQueue canMakePayments]]);
 }
 
+- (void)startProductRequest:(FlutterMethodCall *)call result:(FlutterResult)result {
+  if (![call.arguments isKindOfClass:[NSArray class]]) {
+    result([FlutterError errorWithCode:@"storekit_invalide_argument"
+                               message:@"Argument type of startRequest is not array"
+                               details:call.arguments]);
+    return;
+  }
+  NSArray *productsIdentifiers = (NSArray *)call.arguments;
+  SKProductsRequest *request =
+      [self getProductRequestWithIdentifiers:[NSSet setWithArray:productsIdentifiers]];
+  FIAPRequestHandler *handler = [[FIAPRequestHandler alloc] initWithRequest:request];
+  handler.delegate = self;
+  [self.RequestHandlerSet addObject:handler];
+  [handler startProductRequestWithCompletionHandler:^(SKProductsResponse *_Nullable response,
+                                                      NSError *_Nullable error) {
+    if (error) {
+      NSString *details = [NSString stringWithFormat:@"Reason:%@\nRecoverSuggestion:%@",
+                                                     error.localizedFailureReason,
+                                                     error.localizedRecoverySuggestion];
+      result([FlutterError errorWithCode:error.domain message:error.description details:details]);
```

Comment:
The error code should be probably a value in our control, otherwise there's no way to guarantee we don't report different errors with the same code.